src/runtime/hexagon_remote/sim_host.cpp

```diff
@@ -69,21 +69,30 @@ int init_sim() {
     }
 
     // Configue tracing.
-    const char *T = getenv("HL_HEXAGON_SIM_MIN_TRACE");
-    if (T && T[0] != 0) {
-        status = sim->SetTracing(HEX_TRACE_PC_MIN, T);
+    const char *trace = getenv("HL_HEXAGON_SIM_MIN_TRACE");
+    if (trace && trace[0] != 0) {
```

Comment:
The structure of these if-else is not great, I think it should be rewritten somehow. Note that the structure is like so:

    const char *trace = ...;
    if (trace) { 
         ... 
    } else {
        const char *trace = ...;
        if (trace) {
             ... 
        } else {
            const char *trace = ...;
            if (trace) {
                ...
            }
        }
    }

As you can see, when written like this, the `trace` variable is declared 3 times (shadowed twice), and the level of indentation will simply increase for each new trace flag that gets added.

Are these things really mutually exclusive? Either way, a rewrite to something like the following seems like it would be better:

    struct Trace {
        const char *env_var;
        int hex_trace;
    };
    Trace traces[] = {
        { "HL_HEXAGON_SIM_MIN_TRACE", HEX_TRACE_PC_MIN },
        { "HL_HEXAGON_SIM_TRACE", HEX_TRACE_PC },
        { "HL_HEXAGON_SIM_MEM_TRACE", HEX_TRACE_MEM },
    }
    for (auto i : traces) {
        const char *trace = getenv(i.env_var);
        if (trace && trace[0] != 0) {
            status = sim->SetTracing(i.hex_trace, trace);
            if (status != HEX_STAT_SUCCESS) {
                printf("HexagonWrapper::SetTracing failed: %d\n", status);
                return -1;
            }
        }
    }

If these things are mutually exclusive, add a `break` at the end of `if (trace)`.

Or if you think this is too much logic, *and* the tracing flags aren't mutually exclusive, just copy and past them without the iterated nesting (and rename the trace variable each time so it doesn't collide with others).